### PR TITLE
Modification of the arithmetic operation to use dynamic value rather …

### PR DESCRIPTION
**Pull request by tnicolas29200**
Modification of the arithmetic operation to use dynamic value rather than constants :
the query :
update foo SET id = id + 1
becomes :
update foo SET id = id + #{parameters.p1,jdbcType=INTEGER}

### CODE REVIEW

File: src/test/java/org/mybatis/dynamic/sql/update/UpdateStatementTest.java [hunk 1]
```diff
@@ -171,6 +171,29 @@ public void testFullUpdateStatementNoWhere() {
...
+        String expectedStatement = "update foo " 
+                + "set id = id + #{parameters.p1,jdbcType=INTEGER}, "
+                + "id = id - #{parameters.p2,jdbcType=INTEGER}, "
+                + "id = id * #{parameters.p3,jdbcType=INTEGER}, "
+                + "id = id / #{parameters.p4,jdbcType=INTEGER}";
+                
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(updateStatement.getUpdateStatement()).isEqualTo(expectedStatement);
+        
+            softly.assertThat(updateStatement.getParameters().size()).isEqualTo(0);
```
Comment (jeffgbutler) [inline src/test/java/org/mybatis/dynamic/sql/update/UpdateStatementTest.java]: This is not right.  If there are 4 parameter markers, there should be 4 parameters.

File: src/main/java/org/mybatis/dynamic/sql/update/UpdateDSL.java [hunk 2]
```diff
@@ -141,6 +141,10 @@ public SetClauseFinisher(SqlColumn<T> column) {
             return incrementBy(() -> value);
         }
         
+        public UpdateDSL<R> incrementBy() {
+            return incrementBy(() -> null);
+        }
+
```
Comment (jeffgbutler) [inline src/main/java/org/mybatis/dynamic/sql/update/UpdateDSL.java]: I don't understand the intent here.  Increment by what?

**Comment by tnicolas29200** [comment]
Hello

My goal with this PR is to parametrize the arithmetic operator :
If you want to build a sql query with a dynamic value for increment, you must do :
UPDATE table SET field = field + xxx 
where xxx is the value you want to add.

With my previous implementation, you can just add a constant value to a field and not use a dynamic value, so you must write to query to increment a field by 1 and by 2 (for exemple).

I hope I'm clearer in my explanations and understand where I want to go ?

Maybe my implementation is not the best way to do such thing ?

**Comment by jeffgbutler** [comment]
OK - I get it.  But this PR doesn't accomplish what you want to do.

I'm going to work on improving the expression support in general.  Hopefully will have something to show in a few days.

**Comment by jeffgbutler** [comment]
I've changed how this works in #26.  I removed the incrementBy, decrementBy, etc. methods in favor of reusing the existing add, subtract, etc. methods.

This does not use bind parameters for the constants, it just writes the constant into the generated SQL.  I am not convinced that using a bind parameter is needed for this.  The generated statements are meant to be one time use only - but you could easily write a function to generate the statement based on some parameter if you wanted to do that.

I'm going to close this PR as the code it changes doesn't exist anymore - but the new method is better and more flexible.  You can see the new method here:

https://github.com/mybatis/mybatis-dynamic-sql/blob/master/src/test/java/examples/animal/data/AnimalDataTest.java#L1824-L1829

**Pull request closed without merging**